### PR TITLE
Only add hours that are in the limited time range

### LINKED ISSUE #63: Load data in time range correctly
When we pass `limit` to `load_data()` this function should only consider data in the given time range (`<start_timestapm, start_timestamp + limit>`) and not try to load the amount of data specified by the `limit`.

### PR DESCRIPTION
**Pull request by marusak**
Closes #63

### CODE REVIEW

**Comment by marusak** [comment]
well, "Fixes" and not "Closes"

File: src/app.jsx [hunk 1]
```diff
@@ -739,7 +739,10 @@ class MetricsHistory extends React.Component {
             const init_current_hour = () => {
                 if (!this.data[current_hour])
                     this.data[current_hour] = [];
-                new_hours.add(current_hour);
+
+                // When limit is considered only add hours in this time range
+                if (!limit || load_timestamp + (limit / SAMPLES_PER_H * 3600000) >= current_hour)
```
Comment (martinpitt) [inline src/app.jsx:744]: 3600000 == `MSEC_PER_H`, but moreover `MSEC_PER_H/SAMPLES_PER_H == INTERVAL`, i. e. `limit/INTERVAL` is exactly the number of samples that you want to get. I find that more intuitive.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! Can you please add a quick test for that? Jumping to yesterday should have an empty graph with just the "no data" notification in a freshly booted VM.

File: src/app.jsx (hunk 1 above)
Comment (marusak) [reply src/app.jsx:744]: > 3600000 == MSEC_PER_H

oh right, lets use that

> MSEC_PER_H/SAMPLES_PER_H == INTERVAL

but I need `MSEC_PER_H * SAMPLES_PER_H`

Comment (marusak) [reply src/app.jsx]: changed it a bit, hope now it is better readable

Comment (martinpitt) [reply src/app.jsx]: No, that's not what you do here - operator precedence :-)

Comment (martinpitt) [reply src/app.jsx]: Still more complicated and harder to understand for my taste -- limit/INTERVAL is exactly the "number / (number/seconds) == duration" that you want to get here, and MSEC_PER_H/SAMPLES_PER_H is just derived from INTERVAL. But anyway, ok like that as well. :grin:

Comment (marusak) [reply src/app.jsx]: Took me longer to get it then I am willing to accept but it is there :smile:

**Comment by martinpitt** [comment]
Hm, this changes testBasic(), but it seems it breaks testEvents.. The date selector in the [screenshot](https://logs.cockpit-project.org/logs/pull-68-20200928-115733-3dc02dd5-rhel-8-3/TestApplication-testEvents-rhel-8-3-127.0.0.2-2301-FAIL.png) seems wrong indeed.

File: src/app.jsx (hunk 1 above)
Comment (martinpitt) [reply src/app.jsx]: Oh argh, I concede defeat -- of course INTERVAL is "seconds/sample", not "sample/seconds" -- silly me! :blush:

**Review by martinpitt — APPROVED** [review]
Cheers!
